mlrun/datastore/azure_blob.py

```diff
@@ -71,6 +93,82 @@ def get_storage_options(self):
         )
         return self._sanitize_storage_options(res)
 
+    def do_connect(self, storage_options):
+        """Connect to the BlobServiceClient, using user-specified connection details.
+        Tries connection string first, then credential, then account key, SAS token, and finally anonymous login.
+
+        Raises
+        ------
+        ValueError if none of the connection details are available
```

Comment:
I understand this is copied code, but we have a convention of raising mlrun exceptions in mlrun, such as `MLRunInvalidArgumentError`.